### PR TITLE
use stricter analysis style

### PR DESCRIPTION
**Pull request by truongsinh**
Describe the purpose of the pull request.

#### Enhancements
- Use stricter analysis style to elaborate whenever we cast

### CODE REVIEW

File: lib/src/cache/normalized_in_memory.dart [hunk 1]
```diff
@@ -2,7 +2,7 @@ import 'package:meta/meta.dart';
 import 'package:graphql_flutter/src/utilities/traverse.dart';
 import 'package:graphql_flutter/src/cache/in_memory.dart';
 
-typedef String DataIdFromObject(Object node);
+typedef DataIdFromObject = String Function(Object node);
```
Comment (micimize) [inline lib/src/cache/normalized_in_memory.dart]: I personally prefer the `typedef String DataIdFromObject(Object node);` style, but don't feel strongly. Everything else looks great IMO

**Comment by truongsinh** [comment]
Btw, all the styles are directly taken from https://github.com/flutter/flutter/blob/master/analysis_options.yaml

File: lib/src/cache/normalized_in_memory.dart (hunk 1 above)
Comment (micimize) [reply lib/src/cache/normalized_in_memory.dart]: ah I see, [`- prefer_generic_function_type_aliases`](https://github.com/dart-lang/sdk/blob/master/docs/language/informal/generic-function-type-alias.md)

**Comment by micimize** [comment]
@HofmannZ cool, I don't have merge authorization though, so you'll have to merge it

**Pull request merged**